File: Optimal/Cpp/q289_movezeroes.cpp
//beats 100% time wise

class Solution {
    public:
        void swapnums(int& a, int&b){
            if(&a!=&b){
            a=a^b;
            b=a^b;
            a=a^b;
        }
        }
        void moveZeroes(vector<int>& nums) {
            int i=0;
            int l=nums.size();
            
            
            for(int j=0;j<l;j++){
                if(nums[j]!=0){
                    swapnums(nums[j],nums[i]);
                    i++;
                }
            }
        }
    };

File: MySolutions/Python/q289_movezeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        r = [i for i in nums if i!=0]
        n_zeros = len(nums)-len(r)
        r.extend([0 for i in range(n_zeros)])
        nums[:] = r[:]
